core/timeline_api.php

```diff
@@ -57,12 +57,23 @@ function timeline_events( $p_start_time, $p_end_time, $p_max_events, $p_filter =
 	$t_result = history_query_result( $t_query_options );
 	$t_count = 0;
 
+	$t_previous_history_event = null;
 	while ( $t_history_event = history_get_event_from_row( $t_result, /* $p_user_id */ auth_get_current_user_id(), /* $p_check_access_to_issue */ true ) ) {
 		$t_event = null;
 		$t_user_id = $t_history_event['userid'];
 		$t_timestamp = $t_history_event['date'];
 		$t_issue_id = $t_history_event['bug_id'];
 
+		if ( $t_previous_history_event != null ) {  //check repeated event
+			if (    $t_previous_history_event['userid'] == $t_history_event['userid']
+				 && $t_previous_history_event['date'] == $t_history_event['date']
+				 && $t_previous_history_event['bug_id'] == $t_history_event['bug_id']
+				 && $t_previous_history_event['type'] == $t_history_event['type']	
```

Comment:
Didn't try, but using $t_user_id, $t_timestamp and $t_issue_id should be better in terms of performance than $t_history_event['userid'], $t_history_event['date'] and $t_history_event['bug_id']

You could also introduce `$t_type = $t_history_event['type'];` and use it in your code and the following switch.

Reply:
@atrol I implemented your suggestion